gssapi/tests/test_raw.py

```diff
@@ -734,6 +734,128 @@ def test_display_mech_attr(self):
             display_out.short_desc.should_be(attr[2])
             display_out.long_desc.should_be(attr[3])
 
+    def helper_cb_check(self, req_flags, init_ret_flags_understood,
+                        accept_ret_flags_understood, init_cb, accept_cb,
+                        m_init_ret_flags,
+                        e_init_ret_flags, m_accept_ret_flags,
+                        e_accept_ret_flags):
+        target_name = gb.import_name(TARGET_SERVICE_NAME,
+                                     gb.NameType.hostbased_service)
+
+        server_creds = gb.acquire_cred(None)[0]
+
+        init_ctx = gb.create_sec_context()
+        init_ctx.shouldnt_be_none()
+        init_ctx.should_be_a(gb.SecurityContext)
+
+        gb.set_context_flags(init_ctx, req_flags,
+                             init_ret_flags_understood)
+
+        accept_ctx = gb.create_sec_context()
+        accept_ctx.shouldnt_be_none()
+        accept_ctx.should_be_a(gb.SecurityContext)
+
+        gb.set_context_flags(accept_ctx, req_flags,
+                             accept_ret_flags_understood)
+
+        init_cont_needed = True
+        accept_cont_needed = True
+
+        init_token = None
+        accept_token = None
+
+        init_ret_flags = 0
+        accept_ret_flags = 0
+
+        while True:
+            ctx_resp = None
+            ctx_resp = gb.init_sec_context(target_name=target_name,
+                                           context=init_ctx,
+                                           input_token=accept_token,
+                                           channel_bindings=init_cb)
+            ctx_resp.shouldnt_be_none()
+
+            (init_ctx, _, init_ret_flags, init_token, _,
+             init_cont_needed) = ctx_resp
+
+            if not accept_cont_needed:
+                break
+
+            ctx_resp = gb.accept_sec_context(init_token, context=accept_ctx,
+                                             acceptor_creds=server_creds,
+                                             channel_bindings=accept_cb)
+            ctx_resp.shouldnt_be_none()
+
+            (ctx, _, _, accept_token, accept_ret_flags, _,
+             _, accept_cont_needed) = ctx_resp
+
+            if not init_cont_needed:
+                break
+
+        if ((int(init_ret_flags) & m_init_ret_flags) != e_init_ret_flags):
```

Comment:
shouldn't need a cast to `int` here -- `IntEnumFlagSet` handles that case -- that's why we have it ;-).